scylla-cql/src/deserialize/value.rs

```diff
@@ -902,6 +935,220 @@ where
     }
 }
 
+pub struct ConstLengthVectorIterator<'frame, 'metadata, T> {
+    coll_typ: &'metadata ColumnType<'metadata>,
+    elem_typ: &'metadata ColumnType<'metadata>,
+    remaining: usize,
+    elem_len: usize,
+    slice: FrameSlice<'frame>,
+    phantom_data: std::marker::PhantomData<T>,
+}
+
+impl<'frame, 'metadata, T> ConstLengthVectorIterator<'frame, 'metadata, T> {
+    fn new(
+        coll_typ: &'metadata ColumnType<'metadata>,
+        elem_typ: &'metadata ColumnType<'metadata>,
+        count: usize,
+        elem_len: usize,
+        slice: FrameSlice<'frame>,
+    ) -> Self {
+        Self {
+            coll_typ,
+            elem_typ,
+            remaining: count,
+            elem_len,
+            slice,
+            phantom_data: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<'frame, 'metadata, T> DeserializeValue<'frame, 'metadata>
+    for ConstLengthVectorIterator<'frame, 'metadata, T>
+where
+    T: DeserializeValue<'frame, 'metadata>,
+{
+    fn type_check(typ: &ColumnType) -> Result<(), TypeCheckError> {
+        match typ {
+            ColumnType::Vector { typ: t, .. } => {
+                if t.type_size().is_none() {
+                    return Err(mk_typck_err::<Self>(
+                        typ,
+                        VectorTypeCheckErrorKind::ElementSizeUnknown,
+                    ));
+                }
+                <T as DeserializeValue<'frame, 'metadata>>::type_check(t).map_err(|err| {
+                    mk_typck_err::<Self>(typ, VectorTypeCheckErrorKind::ElementTypeCheckFailed(err))
+                })?;
+                Ok(())
+            }
+            _ => Err(mk_typck_err::<Self>(
+                typ,
+                VectorTypeCheckErrorKind::NotVector,
+            )),
+        }
+    }
+
+    fn deserialize(
+        typ: &'metadata ColumnType<'metadata>,
+        v: Option<FrameSlice<'frame>>,
+    ) -> Result<Self, DeserializationError> {
+        let (t, dim) = match typ {
+            ColumnType::Vector {
+                typ: t,
+                dimensions: dim,
+            } => (t, dim),
+            _ => {
+                unreachable!("Typecheck should have prevented this scenario!")
+            }
+        };
+
+        let v = ensure_not_null_frame_slice::<Self>(typ, v)?;
+
+        Ok(Self::new(typ, t, *dim as usize, t.type_size().unwrap(), v))
+    }
```

Comment:
`t` -> `elem_type` or `element_type`
`dim` -> `dimensions`

Don't be afraid of descriptive variable names.